### PR TITLE
Things needed to be fixed for rails5

### PR DESCRIPTION
**Pull request by bgeuken**
Collection of patches required for Rails 5 that won't break with current rails version

### CODE REVIEW

File: src/api/app/controllers/build_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class BuildController < ApplicationController
+  include ApplicationHelper
```
Comment (hennevogel) [inline src/api/app/controllers/build_controller.rb]: I don't like this at all. Helpers are for views, not to share controller code. ApplicationController is full of methods that can't be called as actions. What's so different about forward_from_backend?

File: src/api/Rakefile [hunk 2]
```diff
@@ -1,8 +1,8 @@
-#!/usr/bin/env rake2.2
+#!/usr/bin/env rake2.3
```
Comment (hennevogel) [inline src/api/Rakefile]: This also needs package updates in OBS:Server:Unstable right?

File: src/api/app/models/bs_request.rb [hunk 3]
```diff
@@ -57,7 +57,7 @@ class SaveError < APIException
 
   after_update :send_state_change
 
-  def save!
+  def save!(args = {})
```
Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: Hm who called this with an argument?

File: src/api/Rakefile (hunk 2 above)
Comment (bgeuken) [inline src/api/Rakefile]: OBS already runs on 2.3. This probably was missed when upgrading ruby, and rake.

File: src/api/app/models/bs_request.rb (hunk 3 above)
Comment (bgeuken) [inline src/api/app/models/bs_request.rb]: Calling << on a CollectionProxy calls our overwritten save, which then calls super... One of the methods expects an argument the other doesn't => ArgumentError

Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: OMG...

Comment (bgeuken) [inline src/api/app/models/bs_request.rb]: What's up?

File: src/api/app/controllers/build_controller.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/app/controllers/build_controller.rb]: Don't like it either...

File: src/api/app/models/bs_request.rb (hunk 3 above)
Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: Just the usual agony ;-)

**Comment by hennevogel** [comment]
LGTM

**Pull request merged**